spec/views/admin_general/_announcement.html.erb_spec.rb

```diff
@@ -0,0 +1,37 @@
+require 'spec_helper'
+
+describe 'admin_general/_announcement.html.erb' do
+
+  def render_view
+    render :partial => 'admin_general/announcement'
+  end
+
+  let(:announcement) do
+    FactoryBot.
+      create(:announcement,
+             visibility: 'admin',
+             content: '<b>Exciting news!</b> <script>alert("!")</script>')
+  end
+
+  describe 'displaying an announcement' do
+    before do
+      assign(:locals, { announcement: announcement } )
+    end
+
+    it 'shows the announcement' do
+      render template: 'admin_general/_announcement',
+             locals: { announcement: announcement }
+      expect(rendered).to match('Introducing projects')
+    end
+
+    it 'strips out dangerous tags' do
+      render template: 'admin_general/_announcement',
+             locals: { announcement: announcement }
+      expect(rendered).to match('<b>Exciting news!</b>')
+      expect(rendered).not_to match('<script>')
+      expect(rendered).not_to match('alert("!")')
```

Comment:
Not sure why this is passing:

![screen shot 2018-10-05 at 11 43 31](https://user-images.githubusercontent.com/282788/46531268-8660b580-c894-11e8-822c-bc09c244ee32.png)

Reply:
ah... I used `match` which take a regex so I'd have needed to say something like `not_to match(/alert\(\"!\"\)/)`\\

so it now reads: `not to include('alert("!")')` as that's easier to make sense of